pkg/resources/collector/logging.go

```diff
@@ -184,3 +184,26 @@ func (r *Reconciler) hostLoggingVolumes() (
 	}
 	return
 }
+
+func (r *Reconciler) hostTraceVolumes() (
+	retVolumeMounts []corev1.VolumeMount,
+	retVolumes []corev1.Volume,
+	retErr error,
+) {
+	retVolumes = append(retVolumes, corev1.Volume{
+		Name: "otel-traces",
+		VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{
+				Path: "/",
+			},
+		},
+	})
+	retVolumeMounts = append(retVolumeMounts, corev1.VolumeMount{
+		Name:             "otel-traces",
+		ReadOnly:         true,
+		MountPath:        otelConfigDir,
+		MountPropagation: lo.ToPtr(corev1.MountPropagationHostToContainer),
+	})
+
+	return
+}
```

Comment:
I could be wrong, but I don't think we need any additional volumes for traces, traces don't require any additional persistent storage. 

I see here that you're attempting to mount a config,  the config for this collector is already in the template written to the config map.  Each of metrics, logs and traces use the same config.

[daemon config map](https://github.com/rancher/opni/blob/98ac856f97a4ecabf8ac5eff31c89d6c5a48c46d/pkg/resources/collector/workloads.go#L113)

[aggregator config map](https://github.com/rancher/opni/blob/98ac856f97a4ecabf8ac5eff31c89d6c5a48c46d/pkg/resources/collector/workloads.go#L152)

[volumes mounting the daemon set config map](https://github.com/rancher/opni/blob/98ac856f97a4ecabf8ac5eff31c89d6c5a48c46d/pkg/resources/collector/workloads.go#L192-L209)

[volumes mounting the aggregator config map](https://github.com/rancher/opni/blob/98ac856f97a4ecabf8ac5eff31c89d6c5a48c46d/pkg/resources/collector/workloads.go#L327-L345)

Reply:
Makes sense. For some reason, my initial understanding was that the collector worked like a database (?). For that reason, I thought we needed a separate volume. Thanks!